kani-compiler/src/kani_middle/stubbing/transform.rs

```diff
@@ -23,18 +26,43 @@ pub fn get_stub(tcx: TyCtxt, def_id: DefId) -> Option<DefId> {
 
 /// Returns the new body of a function/method if it has been stubbed out;
 /// otherwise, returns the old body.
-pub fn transform<'tcx>(
-    tcx: TyCtxt<'tcx>,
-    def_id: DefId,
-    old_body: &'tcx Body<'tcx>,
-) -> &'tcx Body<'tcx> {
+pub fn transform<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId, old_body: &'tcx Body<'tcx>) -> Body<'tcx> {
     if let Some(replacement) = get_stub(tcx, def_id) {
         let new_body = tcx.optimized_mir(replacement).clone();
         if check_compatibility(tcx, def_id, old_body, replacement, &new_body) {
-            return tcx.arena.alloc(new_body);
+            return new_body;
+        }
+    }
+    old_body.clone()
+}
+
+struct ExternFunctionTransformer<'tcx> {
+    tcx: TyCtxt<'tcx>,
+    body: Body<'tcx>,
+}
+
+impl<'tcx> MutVisitor<'tcx> for ExternFunctionTransformer<'tcx> {
+    fn tcx(&self) -> TyCtxt<'tcx> {
+        self.tcx
+    }
+
+    fn visit_operand(&mut self, operand: &mut Operand<'tcx>, _location: Location) {
+        let func_ty = operand.ty(&self.body, self.tcx);
+        if let ty::FnDef(reachable_function, generics) = *func_ty.kind() {
+            if let Some(stub) = get_stub(self.tcx, reachable_function) {
+                let Operand::Constant(fn_def) = operand else { unreachable!() };
```

Comment:
This can actually fail if the user uses function pointers so I'd recommend to do one of the following:

1. *We disallow function pointers:* Then this should be a graceful `self.tcx.span_err`
2. *We allow function pointers:* Then in this case you just ignore this operand, since you know the visitor will have performed the stub swap earlier where the place in the operand was assigned.

Reply:
I believe @feliperodri is still working on enabling function pointers

Reply:
I think with this it could just work, since the visitor checks every kind of operand, including in assignments so it should swap them out even there.

Reply:
The current implementation already takes care of function pointers. I added a test case on this. Am I missing something?